tests/transforms/test_split_non_commuting.py

```diff
@@ -95,7 +92,7 @@ def complex_qwc_processing_fn(results):
     return (
         group0[0],
         0.5 * group1[0],
-        group0[0] + group1[1] + 2.0 * group2[0] + 1.0,
+        group0[0] + group1[1] + 2.0 * group2 + 1.0,
```

Comment:
Is there any harm in `group2[0]` to match the others, or is it outright wrong?

Reply:
It's outright wrong, because `group2` corresponds to a tape that only has one measurement. This dimension should've been squeezed out.

Reply:
We squeeze out the single dimension of the results when there is a single measurement process.